module/state_synchronization/indexer/collection_executed_metric.go

```diff
@@ -87,17 +95,41 @@ func (c *CollectionExecutedMetricImpl) BlockFinalized(block *flow.Block) {
 		}
 
 		for _, t := range l.Transactions {
+			c.blockTransactions[blockID] = append(c.blockTransactions[blockID], t)
 			c.accessMetrics.TransactionFinalized(t, now)
 		}
 	}
 
+	// Process block seals
+	for _, s := range block.Payload.Seals {
+		if transactions, found := c.blockTransactions[s.BlockID]; found && len(transactions) != 0 {
+			for _, t := range transactions {
+				c.accessMetrics.TransactionSealed(t, now)
+
+				// Remove the transaction by transaction ID
+				c.blockTransactions[s.BlockID] = removeTransactionByID(c.blockTransactions[s.BlockID], t)
```

Comment:
can we just remove the whole block set after processing instead of each transaction one at a time?